app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.kt

```diff
@@ -0,0 +1,1273 @@
+        when (key) {
+            "fabHide" -> {
+                hideFab = sharedPreferences.getBoolean("fabHide", false)
+            }
+            "mediaPreviewEnabled" -> {
+                val enabled = accountManager.activeAccount!!.mediaPreviewEnabled
+                val oldMediaPreviewEnabled = adapter.mediaPreviewEnabled
+                if (enabled != oldMediaPreviewEnabled) {
+                    adapter.mediaPreviewEnabled = enabled
+                    fullyRefresh()
+                }
+            }
+            "tabFilterHomeReplies" -> {
+                val filter = sharedPreferences.getBoolean("tabFilterHomeReplies", true)
+                val oldRemoveReplies = filterRemoveReplies
+                filterRemoveReplies = kind == Kind.HOME && !filter
+                if (adapter.itemCount > 1 && oldRemoveReplies != filterRemoveReplies) {
+                    fullyRefresh()
+                }
+            }
+            "tabFilterHomeBoosts" -> {
+                val filter = sharedPreferences.getBoolean("tabFilterHomeBoosts", true)
+                val oldRemoveReblogs = filterRemoveReblogs
+                filterRemoveReblogs = kind == Kind.HOME && !filter
+                if (adapter.itemCount > 1 && oldRemoveReblogs != filterRemoveReblogs) {
+                    fullyRefresh()
+                }
+            }
+            Filter.HOME, Filter.NOTIFICATIONS, Filter.THREAD, Filter.PUBLIC, Filter.ACCOUNT -> {
+                if (filterContextMatchesKind(kind, listOf(key))) {
+                    reloadFilters(true)
+                }
+            }
+            "alwaysShowSensitiveMedia" -> {
+                //it is ok if only newly loaded statuses are affected, no need to fully refresh
+                alwaysShowSensitiveMedia = accountManager.activeAccount!!.alwaysShowSensitiveMedia
+            }
+        }
+    }
+
+    public override fun removeItem(position: Int) {
+        statuses.removeAt(position)
+        updateAdapter()
+    }
+
+    private fun removeAllByAccountId(accountId: String) {
+        // using iterator to safely remove items while iterating
+        val iterator = statuses.iterator()
+        while (iterator.hasNext()) {
+            val status = iterator.next().asRightOrNull()
+            if (status != null &&
+                    (status.account.id == accountId || status.actionableStatus.account.id == accountId)) {
+                iterator.remove()
+            }
+        }
+        updateAdapter()
+    }
+
+    private fun removeAllByInstance(instance: String) {
+        // using iterator to safely remove items while iterating
+        val iterator = statuses.iterator()
+        while (iterator.hasNext()) {
+            val status = iterator.next().asRightOrNull()
+            if (status != null && LinkHelper.getDomain(status.account.url) == instance) {
+                iterator.remove()
+            }
+        }
+        updateAdapter()
+    }
```

Comment:
I wanted to rewrite these with removeIf like so:

```kotlin
    private fun removeAllByAccountId(accountId: String) {
        // using iterator to safely remove items while iterating
        statuses.removeIf { either ->
            val status = either.asRightOrNull()
            if (status != null) {
                status.account.id == accountId || status.actionableStatus.account.id == accountId
            } else {
                false
            }
        }
        updateAdapter()
    }

    private fun removeAllByInstance(instance: String) {
        // using iterator to safely remove items while iterating
        statuses.removeIf { either ->
            val status = either.asRightOrNull()
            status != null && LinkHelper.getDomain(status.account.url) == instance
        }
        updateAdapter()
    }
```

but it needs API 24. I don't know if it's handled by compile transformations these days or not, I remember they ported some APIs

Reply:
We need to enable desugaring to use it https://developer.android.com/studio/write/java8-support#library-desugaring

The only drawback is a little bit more app size and build time, but I guess its worth it?

Reply:
Probably doesn't make too much sense as we are using Kotlin for most things. Here it doesn't work because `PairedList` is our own list in Java I assume.

Reply:
kotlin has `removeAll`:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/remove-all.html